tests/test_spi_devices.py

```diff
@@ -0,0 +1,339 @@
+from __future__ import (
+    unicode_literals,
+    absolute_import,
+    print_function,
+    division,
+    )
+str = type('')
+
+
+import sys
+import pytest
+from collections import namedtuple
+try:
+    from math import isclose
+except ImportError:
+    from gpiozero.compat import isclose
+
+from gpiozero import *
+from gpiozero.pins.mock import MockSPIDevice, MockPin
+
+
+def setup_function(function):
+    import gpiozero.devices
+    gpiozero.devices.pin_factory = MockPin
+
+def teardown_function(function):
+    MockPin.clear_pins()
+
+def clamp(v, min_value, max_value):
+    return min(max_value, max(min_value, v))
+
+def scale(v, ref, bits):
+    v /= ref
+    vmin = -(2 ** bits)
+    vmax = -vmin - 1
+    vrange = vmax - vmin
+    return int(((v + 1) / 2.0) * vrange + vmin)
+
+
+class MockMCP3xxx(MockSPIDevice):
+    def __init__(
+            self, clock_pin, mosi_pin, miso_pin, select_pin=None,
+            channels=8, bits=10):
```

Comment:
I'm not sure if it makes sense to provide default-values for the `channels` and `bits` args?
But it obviously makes sense to have them after the optional `select_pin` arg, so maybe you could default them to None, and then assert if they're still None, like you do in RGBLED? https://github.com/RPi-Distro/python-gpiozero/blob/master/gpiozero/output_devices.py#L556